src/test/inferencerequest_test.cpp

```diff
@@ -58,7 +58,40 @@ const OVMS_DataType DATATYPE{OVMS_DATATYPE_FP32};
 TEST(InferenceParameter, CreateParameter) {
     InferenceParameter parameter(PARAMETER_NAME.c_str(), PARAMETER_DATATYPE, reinterpret_cast<const void*>(&PARAMETER_VALUE));
 }
-
+TEST(Buffer, StringHandling) {
+    using std::string;
+    using vs_t = std::vector<string>;
+    const string* sptr{nullptr};
+    vs_t intelText{{"Intel"}, {"owns"}, {"OVMS"}};
+    vs_t nvidiaText{{"NVIDIA"}, {"owns"}, {"Triton"}};
+    std::unique_ptr<const Buffer> bufferWithCopy{nullptr};
+    const void* ptr{nullptr};
+    {
+        vs_t text2BeDeleted = nvidiaText;
+        const Buffer bufferWithNoCopy(&intelText[0], intelText.size() * sizeof(string), OVMS_BUFFERTYPE_CPU);
+        auto vecstruniqptr = std::make_unique<vs_t>(std::move(text2BeDeleted));
+        bufferWithCopy = std::make_unique<const Buffer>(std::move(vecstruniqptr));
+        // check with nocopy
+        ptr = bufferWithNoCopy.data();
+        EXPECT_EQ(intelText.data(), &intelText[0]);
+        EXPECT_EQ(ptr, &intelText[0]);
+        sptr = reinterpret_cast<const string*>(ptr);
+        EXPECT_EQ(intelText.size(), bufferWithNoCopy.getByteSize() / sizeof(string));
+        EXPECT_TRUE(std::equal(intelText.begin(), intelText.end(), sptr));
+        // now check with copy
+        ptr = bufferWithCopy->data();
+        sptr = reinterpret_cast<const string*>(ptr);
+        EXPECT_EQ(nvidiaText.size(), bufferWithNoCopy.getByteSize() / sizeof(string));
+        EXPECT_TRUE(std::equal(nvidiaText.begin(), nvidiaText.end(), sptr));
+    }
+    // now text is deleted but for buffer with copy still expect to work
+    vs_t randomData{{"Intel"}, {"owns"}, {"DCAI"}};
+    ptr = bufferWithCopy->data();
+    sptr = reinterpret_cast<const string*>(ptr);
+    EXPECT_EQ(bufferWithCopy->getByteSize(), sizeof(string) * nvidiaText.size());
```

Comment:
should we also check if the content remains unchanged?

Reply:
There is full scope tests with C-API which does that but I added it here as well.